### PR TITLE
Add `highest_out` to `create_constraints_indices`

### LINKED ISSUE #914: Create constraints indices for `highest_out`
### Description

Similar to what has been done in #896.

### PR DESCRIPTION
**Pull request by gnawin**
## Related issues

Closes #914 

## Checklist

- [x] I am following the [contributing guidelines](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/blob/main/docs/src/90-contributing.md)

- [x] Tests are passing
- [ ] Lint workflow is passing
- [x] Docs were updated and workflow is passing

### CODE REVIEW

**Comment by gnawin** [comment]
@abelsiqueira could you help on this? thanks.

File: src/tmp.jl [hunk 1]
```diff
@@ -231,6 +231,54 @@ function tmp_create_constraints_indices(connection)
...
+    # FROM asset_time_resolution
+    # UNION
+    DuckDB.execute(
+        connection,
+        "CREATE OR REPLACE TABLE cons_indices_highest_out AS
+        SELECT
+            main.asset,
+            main.year,
+            main.rep_period,
+            sub.time_block_start,
```
Comment (abelsiqueira) [inline src/tmp.jl:267]:
```suggestion
            COALESCE(sub.time_block_start, 1) AS time_block_start,
```
I think this is the same issue that I had in #919 but for the benchmark tests, and that now is fixed in #920. The problem is that there are no outgoing flows (in this case), so `sub.time.block_start` doesn't exist. By using `COALESCE`, we make sure that if it doesn't exist, we use the range `1:num_timesteps`

**Comment by gnawin** [comment]
@abelsiqueira the tests indeed passed, but the link checker complains. Could you have a look? Or pause/close this. Thank you!

**Review by abelsiqueira — APPROVED** [review]
LGTM. Benchmarks are not running because it's a fork, but should be a similar efficiency to previous PR (slight decrease with possibilities of improvements).

**Pull request merged**